napalm_panos/panos.py

```
@@ -534,3 +534,107 @@ def get_interfaces(self):
+
+            v4_ip = parsed_intf_dict.get('ip')
+            secondary_v4_ip = parsed_intf_dict.get('addr')
+            v6_ip = parsed_intf_dict.get('addr6')
+
+            if v4_ip != 'N/A':
+                address, pref = v4_ip.split('/')
+                _ip_info[intf].setdefault('ipv4', {})[address] = {'prefix_length': int(pref)}
+
+            if secondary_v4_ip is not None:
+                members = secondary_v4_ip['member']
+                if not isinstance(members, list):
+                    # If only 1 secondary IP is present, xmltodict converts field to a string, else
+                    # it converts it to a list of strings.
+                    members = [members]
+                for address in members:
+                    address, pref = address.split('/')
+                    _ip_info[intf].setdefault('ipv4', {})[address] = {'prefix_length': int(pref)}
+
+            if v6_ip is not None:
+                members = v6_ip['member']
+                if not isinstance(members, list):
+                    # Same "1 vs many -> string vs list of strings" comment.
+                    members = [members]
+                for address in members:
+                    address, pref = address.split('/')
+                    _ip_info[intf].setdefault('ipv6', {})[address] = {'prefix_length': int(pref)}
+
+            # Reset dictionary if no addresses were found.
+            if _ip_info == {intf: {}}:
+                _ip_info = {}
+
+            return _ip_info
+
+        ip_interfaces = {}
+        cmd = "<show><interface>all</interface></show>"
+
+        try:
+            self.device.op(cmd=cmd)
+            interface_info_xml = xmltodict.parse(self.device.xml_root())
+            interface_info_json = json.dumps(
+                interface_info_xml['response']['result']['ifnet']['entry']
+            )
+            interface_info = json.loads(interface_info_json)
+        except KeyError:
+            return ip_interfaces
```

not following this one? When would `<show><interface>all</interface></show>` return a condition that meets this?

---

good point, I was following the original implementation which had a similar approach, but missed giving that some thought.

Seems the better approach would be remove the `try/except` altogether, and let the call fail if something goes wrong. Come to think of it that should IMO be the same approach for the new `get_interfaces()` too.